test/helpers/misc.rb

```diff
@@ -108,3 +108,12 @@ def stringify_keys_in_object(object)
     object
   end
 end
+
+def wait_until_not_nil(give_up_after = 3, &block)
+  total_tries = give_up_after * 10
```

Comment:
Can we pass `total_tries` as the argument and call it like `wait_until_not_nil(50) { ... }`? Or is there a reason for the *10 here?

Reply:
I think the idea is so that we can pass time (seconds) rather than tries to know how long it will take. Happy to change it to pass `total_tries` if that's what is preferred!